arch/arm64/src/qemu/qemu_serial.c

```diff
@@ -0,0 +1,856 @@
+ * Name: qemu_pl011_shutdown
+ *
+ * Description:
+ *   Disable the UART.  This method is called when the serial
+ *   port is closed
+ *
+ ***************************************************************************/
+
+static void qemu_pl011_shutdown(struct uart_dev_s *dev)
+{
+  sinfo("%s: call unexpected\n", __func__);
+}
+
+static int qemu_pl011_setup(struct uart_dev_s *dev)
+{
+  struct pl011_uart_port_s  *sport  = (struct pl011_uart_port_s *)dev->priv;
+  const struct pl011_config *config = &sport->config;
+  struct pl011_data         *data   = &sport->data;
+  int                       ret;
+  uint32_t                  lcrh;
+  irqstate_t                i_flags;
+
+  i_flags =  up_irq_save();
+
+  /* If working in SBSA mode, we assume that UART is already configured,
+   * or does not require configuration at all (if UART is emulated by
+   * virtualization software).
+   */
+
+  if (!data->sbsa)
+    {
+      /* disable the uart */
+
+      pl011_disable(sport);
+      pl011_disable_fifo(sport);
+
+      /* Set baud rate */
+
+      ret = pl011_set_baudrate(sport, config->sys_clk_freq,
+                            data->baud_rate);
+      if (ret != 0)
+        {
+          up_irq_restore(i_flags);
+          return ret;
+        }
+
+      /* Setting the default character format */
+
+      lcrh  = config->uart->lcr_h & ~(PL011_LCRH_FORMAT_MASK);
+      lcrh  &= ~(BIT(0) | BIT(7));
+      lcrh  |= PL011_LCRH_WLEN_SIZE(8) << PL011_LCRH_WLEN_SHIFT;
+      config->uart->lcr_h = lcrh;
+
+      /* Enabling the FIFOs */
+
+      pl011_enable_fifo(sport);
+    }
+
+  /* initialize all IRQs as masked */
+
+  config->uart->imsc    = 0U;
+  config->uart->icr     = PL011_IMSC_MASK_ALL;
+
+  if (!data->sbsa)
+    {
+      config->uart->dmacr = 0U;
+      __ISB();
+      config->uart->cr  &= ~(BIT(14) | BIT(15) | BIT(1));
+      config->uart->cr  |= PL011_CR_RXE | PL011_CR_TXE;
+      __ISB();
+    }
+
+  up_irq_restore(i_flags);
+
+  return 0;
+}
+
+/***************************************************************************
+ * Private Data
+ ***************************************************************************/
+
+/* Serial driver UART operations */
+
+static const struct uart_ops_s g_uart_ops =
+{
+  .setup    = qemu_pl011_setup,
+  .shutdown = qemu_pl011_shutdown,
+  .attach   = qemu_pl011_attach,
+  .detach   = qemu_pl011_detach,
+  .ioctl    = qemu_pl011_ioctl,
+  .receive  = qemu_pl011_receive,
+  .rxint    = qemu_pl011_rxint,
+  .rxavailable = qemu_pl011_rxavailable,
+#ifdef CONFIG_SERIAL_IFLOWCONTROL
+  .rxflowcontrol    = NULL,
+#endif
+  .send     = qemu_pl011_send,
+  .txint    = qemu_pl011_txint,
+  .txready  = qemu_pl011_txready,
+  .txempty  = qemu_pl011_txempty,
+};
+
+/* This describes the state of the uart1 port. */
+
+static struct pl011_uart_port_s g_uart1priv =
+{
+  .data             =
+  {
+    .baud_rate      = CONFIG_UART1_BAUD, .sbsa       = false,
```

Comment:
```suggestion
    .baud_rate      = CONFIG_UART1_BAUD,
    .sbsa           = false,
```

Reply:
Oh, I use /tools/uncrustify.cfg in VSCode with uncrustify extention,
but format like this ugly style
thanks